### PR TITLE
Add Customer and Organization tax numbers

### PR DESCRIPTION
**Pull request by VladimirBerdnik**
## Summary
This PR will add tax numbers to customers and organizations to associate with invoices.

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/Common/CommonInvoice.yaml [hunk 1]
```diff
@@ -58,6 +58,32 @@ properties:
...
+    type: object
+    allOf:
+      - $ref: ../ResourceId.yaml
+  customerTaxNumber:
+    description: Customer tax number associated with the invoice.
+    readOnly: true
+    nullable: true
+    type: object
+    allOf:
+      - $ref: ../TaxNumber.yaml
```
Comment (ashkarpetin) [inline openapi/components/schemas/Common/CommonInvoice.yaml:86]: I think these properties should not be in the root of invoice schema

File: openapi/components/schemas/Common/CommonInvoice.yaml [hunk 2]
```diff
@@ -58,6 +58,32 @@ properties:
     $ref: ../Shipping.yaml
   tax:
     $ref: ../Invoices/InvoiceTax.yaml
+  organizationTaxNumberId:
+    description: Tax number identifier to associate with the invoice.
```
Comment (subenson) [inline openapi/components/schemas/Common/CommonInvoice.yaml:62]: I don't understand the difference between taxNumber and taxNumberId for both organization and customer. Can you elaborate?

File: openapi/components/schemas/TaxNumber.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  id:
+    description: Unique tax number identifier.
+    readOnly: true
+    allOf:
+      - $ref: ./ResourceId.yaml
+  type:
+    type: string
+    description: Type of the tax number.
+    enum:
+      - eu-vat
```
Comment (mehicalen) [inline openapi/components/schemas/TaxNumber.yaml:15]: Here is the list of all tax IDs I scraped from [here](https://stripe.com/docs/billing/customer/tax-ids).

Link: https://github.com/Rebilly/frontend-recomm/pull/4116/files#diff-c53e82e0384d42b5793b337b58c34b9da3a6c7d3e1b34a4c6c62cb7b4362bf3c

File: openapi/components/schemas/Common/CommonInvoice.yaml (hunk 1 above)
Comment (VladimirBerdnik) [reply openapi/components/schemas/Common/CommonInvoice.yaml]: removed

**Comment by VladimirBerdnik** [comment]
devs please review

File: openapi/components/schemas/TaxNumber.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+type: object
+required:
+  - type
+  - value
+properties:
+  id:
+    description: Unique tax number identifier.
```
Comment (noon-dawg) [inline openapi/components/schemas/TaxNumber.yaml:7]: suggestion:  
```suggestion
    description: Unique tax identification number.
```

**Review by lexerom — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Pull request merged**